src/create_json.py

```diff
@@ -60,6 +61,22 @@ def remove_duplicate_dicts(l: list[dict]) -> list[dict]:
     return new_l
 
 
+def null_empty_exam_dates(timetable: pd.DataFrame) -> None:
+    """
+    Function to insert 'null' in exams dates where the course has no exam date to
+    prevent wrong exam date being copied due to ffill().
+
+    Args:
+        timetable (pd.DataFrame): The timetable dataframe to fix dates.
+    """
+    for index, row_data in timetable.iterrows():
+        if pd.notna(row_data[0]) and not (pd.notna(row_data[-2])):
+            timetable.loc[index, "midsem"] = "null"
+
+        if pd.notna(row_data[0]) and not (pd.notna(row_data[-1])):
+            timetable.loc[index, "compre"] = "null"
```

Comment:
we need the JS `null` not the string `"null"`. iinw `json.dumps` converts python `None` to a js/json `null`.